src/hooks/__tests__/useApi.test.ts

```diff
@@ -0,0 +1,44 @@
+import { renderHook } from '@testing-library/react-hooks';
+
+import useApi from '../useApi';
+
+describe('useApi', () => {
+  it('returns empty error and result while fetching', () => {
+    // we use a promise here that never returns anything, so the hook never returns a result
+    jest.spyOn(window, 'fetch').mockImplementationOnce(() => new Promise<Response>(() => { }));
+    const { result } = renderHook(() => useApi('/foo/loading'));
+    const [error, apiResult] = result.current;
+
+    expect(error).toBeNull();
+    expect(apiResult).toBeNull();
+  });
+
+  it('returns response JSON if request succeeds', async () => {
+    const expectedResult = { test: 42 };
+    const response = new Response(JSON.stringify(expectedResult));
+    jest.spyOn(window, 'fetch').mockResolvedValue(response);
```

Comment:
Here we mock `window.fetch` to return a successful response by using `mockResolvedValue()` (in contrast to `mockRejectedValue()`, which we would use for errors in asynchronous functions). Note that the `resolve` and `reject` relates to the `Promise.resolve()` and `Promise.reject()` functions which are used in conjunction with asynchronous methods.

Reply:
Additional note: We scarcely use `Promise.resolve` and `Promise.reject` directly, we will mostly use `async` and `await`, which translate to promises under the hood, but are a bit easier to work with. Maybe we can have an in-depth session for this later on, for now it's not that important.